src/test/java/pl/apanowicz/demoapp/api/ProductEndpointTests.java

```diff
@@ -0,0 +1,149 @@
+package pl.apanowicz.demoapp.api;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import pl.apanowicz.demoapp.DemoappApplicationTests;
+import pl.apanowicz.demoapp.domain.ProductFacade;
+import pl.apanowicz.demoapp.domain.ProductRequestDto;
+import pl.apanowicz.demoapp.domain.ProductResponseDto;
+
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class ProductEndpointTests extends DemoappApplicationTests {
+
+    @Autowired
+    ProductFacade productFacade;
+
+    @Test
+    public void shouldGetExistingProduct() {
+        ProductRequestDto requestDto = new ProductRequestDto("product");
+        ProductResponseDto existingProduct = productFacade.create(requestDto);
+        final String url = "http://localhost:" + this.port + "/api/v1/products/" + existingProduct.getId();
+
+        ResponseEntity<ProductResponseDto> result = httpClient.getForEntity(url, ProductResponseDto.class);
+
+        assertThat(result.getStatusCodeValue()).isEqualTo(200);
+        assertThat(result.getBody()).isEqualToComparingFieldByField(existingProduct);
+    }
+
+    @Test
+    public void shouldReturnStatus404WhenGetAndProductNotExist() {
+        ProductRequestDto requestDto = new ProductRequestDto("product");
+        final String url = "http://localhost:" + this.port + "/api/v1/products/" + UUID.randomUUID();
```

Comment:
Testy z reguły mają być deterministyczne. W tym przypadku raczej nie powinno to stanowić problemu, ale dobrą praktyką jest korzystanie z ustalonych w testach wartości, a nie losowych. Może warto tutaj ustalić jakąś stałą wartość id produktu, np.
`String productId = "fake_product`?

Reply:
Zagalopowałem się. Poprawiłem. Dzięki ;)